examples/hitechnic-superpro-bar-display.py

```diff
@@ -0,0 +1,31 @@
+from time import sleep
+import nxt.locator
+from nxt.sensor.hitechnic import SuperPro
+
+"""
+For this demo, attach LED's to the HiTechnic SuperPro from pins B0-B7 to GND.
+No resistors necessary, as these pins already have 220 ohm resistors.
+After electrical connections are completed, connect the SuperPro to Port S1 on the NXT.
+
+Note: The original kit only comes with 6 green/red LED's, so if you're short on LED's just connect B0-B5.
+WARNING: If you are light sensitive, avoid this demo. It flashes the B0 LED at ~5Hz.
+"""
+
+# Find NXT, configure sensor
+brick = nxt.locator.find()
+pro = SuperPro(brick, nxt.sensor.Port.S1)
```

Comment:
I ended up leaving the sensor code the way it was because it was causing type errors (?), however I did replace the calls with the `with` statement as suggested.

```python
with nxt.locator.find() as brick:
    # Rest of the code
```